models/classes/TestSessionMetaData.php

```diff
@@ -139,8 +142,58 @@ public function registerItemCallbacks()
                     $itemSession->registerCallback(
                         'suspend',
                         function ($item, $testSessionMetaData) {
+                            $time = microtime(true);
+                            /** @var $testSessionMetaData TestSessionMetaData */
+
+                            $itemResults = $testSessionMetaData->getItemResults();
+
+                            foreach ($itemResults['data'] as $itemResult) {
+                                if (is_array($itemResult) && isset( $itemResult['sortedVars'] ) && isset( $itemResult['sortedVars']['taoResultServer_models_classes_TraceVariable'] )) {
+                                    $sectionResult[] = $testSessionMetaData->getItemServerTime($itemResult['sortedVars']['taoResultServer_models_classes_TraceVariable']);;
+                                }
+                            }
+
+                            $session  = $testSessionMetaData->getTestSession();
+                            $section  = $session->getCurrentAssessmentSection();
+                            $testPart = $session->getCurrentTestPart();
+                            $test     = $session->getAssessmentTest();
+
+                            $testMaxTimeAllowed     = $test->hasTimeLimits() ? $testPart->getTimeLimits()->getMaxTime() : null;
```

Comment:
@antoinerobin as far as i understand issue is related to QtiSDK, but unfortunately i have not permissions to create new branch with fix